launch_ros/launch_ros/event_handlers/on_state_transition.py

```diff
@@ -51,8 +50,19 @@ def __init__(
 
         If matcher is given, the other conditions are not considered.
         """
-        if not isinstance(target_lifecycle_node, (LifecycleNode, type(None))):
-            raise RuntimeError("OnStateTransition requires a 'LifecycleNode' action as the target")
+        lifecycle_property = type(target_lifecycle_node).__dict__.get('is_lifecycle_node', None)
```

Comment:
Just curious, but is `getattr(target_lifecycle_node, 'is_lifecycle_node', None)` not sufficient? https://docs.python.org/3/library/functions.html#getattr

Reply:
The `__dict__` method checks the class definition if it has that attribute rather than the individual instance passed into the function. I thought this was a more strict type-checking method of handling it